src/components/Modals/AddLiquidityModal.js

```diff
@@ -120,119 +126,166 @@ function AddLiquidityModal({
     const decimalsOut = isAsset1 ? decimals2 : decimals1;
     const formatedValue = formatAssets(tradeData, decimalsOut);
     const sanitizedValue = sanitizeValue(formatedValue.toString());
-    setValue('amount1Desired', isAsset1 ? value : sanitizedValue);
-    setValue('amount2Desired', isAsset1 ? sanitizedValue : value);
-    await trigger();
+    if (isAsset1) {
+      form.setFieldValue('amount2Desired', sanitizedValue);
+    } else {
+      form.setFieldValue('amount1Desired', sanitizedValue);
+    }
+    form.validateFields();
   };
 
+  const amount1Desired = Form.useWatch('amount1Desired', form);
+  const amount2Desired = Form.useWatch('amount2Desired', form);
+  const details = Form.useWatch('details', form);
+
+  useEffect(() => {
+    if (asset1Focused && amount1Desired) {
+      handleChangeInput(amount1Desired, true);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [amount1Desired]);
```

Comment:
Are you sure, you want to skip in dependecies asset1Focused?

Reply:
Yes, that is on purpose, this prevents additional useless rerenders